src/main/java/fhirspark/adapter/FollowUpAdapter.java

```diff
@@ -0,0 +1,274 @@
+                Boolean val = obs.getValueBooleanType().getValue();
+
+                try {
+                    ResponseCriteria.class
+                        .getDeclaredMethod("set" + tag, Boolean.class)
+                        .invoke(respCrit, val);
+                } catch (NoSuchMethodException e) {
+                    e.printStackTrace();
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                } catch (InvocationTargetException e) {
+                    e.printStackTrace();
+                }
+
+            } else if (obs.getIdentifierFirstRep().getValue().startsWith(patientId + "_")) {
+
+                therapyRecommendation = TherapyRecommendationAdapter.toJson(client,
+                        regex, obs);
+
+            }
+
+        }
+
+        followUp.setTherapyRecommendation(therapyRecommendation);
+        followUp.setResponse(respCrit);
+
+        return followUp;
+    }
+
+    public static void fromJson(Bundle bundle, List<Regex> regex, Reference fhirPatient,
+        String patientId, FollowUp followUp) {
+
+        MedicationStatement medicationStatement = new MedicationStatement();
+        medicationStatement.getMeta().addProfile(GenomicsReportingEnum.GENOMICS_REPORT.getSystem());
+        medicationStatement.getMeta().addProfile(MolekulargenetischerBefundberichtEnum.GENOMICS_REPORT.getSystem());
+        medicationStatement.addIdentifier().setSystem(followUpUri).setValue(followUp.getId());
+        medicationStatement.setId(IdType.newRandomUuid());
+        medicationStatement.setSubject(fhirPatient);
+
+        medicationStatement.setInformationSource(getOrCreatePractitioner(bundle, followUp.getAuthor()));
+
+        medicationStatement.addReasonReference(
+            getTherapyRecommendationReference(bundle, followUp.getAuthor(), followUp.getTherapyRecommendation().getId())
+        );
+
+        medicationStatement.getEffectiveDateTimeType().fromStringValue(followUp.getDate());
+
+        medicationStatement.addIdentifier().setValue(followUp.getId());
+
+        if (followUp.getComment() != null) {
+            ArrayList<Annotation> annots = new ArrayList<Annotation>();
+            Annotation comment = new Annotation();
+            comment.setText(followUp.getComment());
+            annots.add(comment);
+            medicationStatement.setNote(annots);
+        }
+
+        if (followUp.getTherapyRecommendationRealized()) {
+            medicationStatement.setStatus(MedicationStatement.MedicationStatementStatus.fromCode("completed"));
+        } else {
+            medicationStatement.setStatus(MedicationStatement.MedicationStatementStatus.fromCode("not-taken"));
+        }
+
+        if (followUp.getSideEffect()) {
+            Coding coding = new Coding();
+            CodeableConcept cc = new CodeableConcept();
+            coding.setSystem("http://hl7.org/fhir/ValueSet/reason-medication-status-codes");
```

Comment:
This is wrong. `reason-medication-status-codes` describes a value set, the corresponding code is from SNOMED CT, so the system would be `http://snomed.info/sct`.